apps/rush/src/taskRunner/ProjectBuildTask.ts

```diff
@@ -67,44 +76,59 @@ export default class ProjectBuildTask implements ITaskDefinition {
 
   public execute(writer: ITaskWriter): Promise<TaskStatus> {
     return new Promise<TaskStatus>((resolve: (status: TaskStatus) => void, reject: (errors: TaskError[]) => void) => {
-      let deps: IPackageDeps;
-      PackageChangeAnalyzer.rushConfig = this._rushConfiguration;
+      let build: IBuildCommand;
       try {
-        deps = PackageChangeAnalyzer.instance.getPackageDepsHash(this._rushProject.packageName);
+         build = this._getBuildCommand();
       } catch (error) {
-        writer.writeLine('Unable to calculate incremental build state. ' +
-          'Instead running full rebuild. ' + error.toString());
-        deps = undefined;
+        reject(error);
+        return;
       }
-      this._executeTask(writer, deps, resolve, reject);
+
+      const deps: IPackageDependencies = this._getPackageDependencies(build, writer);
+      this._executeTask(build, writer, deps, resolve, reject);
     });
   }
 
+  private _getPackageDependencies(build: IBuildCommand, writer: ITaskWriter): IPackageDependencies {
+    let deps: IPackageDependencies = undefined;
+    PackageChangeAnalyzer.rushConfig = this._rushConfiguration;
+    try {
+      deps = {
+        files: PackageChangeAnalyzer.instance.getPackageDepsHash(this._rushProject.packageName).files,
+        arguments: `${build.command} ${build.args.join(' ')}`
+      };
+    } catch (error) {
+      writer.writeLine('Unable to calculate incremental build state. ' +
+        'Instead running full rebuild. ' + error.toString());
+    }
+    return deps;
+  }
+
   private _executeTask(
+    build: IBuildCommand,
     writer: ITaskWriter,
-    currentPackageDeps: IPackageDeps,
+    currentPackageDeps: IPackageDependencies,
     resolve: (status: TaskStatus) => void,
     reject: (errors: TaskError[]) => void
   ): void {
     this._hasWarningOrError = false;
 
     const projectFolder: string = this._rushProject.projectFolder;
-    const currentDepsPath: string =
-      path.join(this._rushProject.projectFolder, RushConstants.packageDepsFilename);
-    let lastPackageDeps: IPackageDeps;
+    let lastPackageDeps: IPackageDependencies;
 
     try {
       writer.writeLine(`>>> ${this.name}`);
 
+      const currentDepsPath: string = path.join(this._rushProject.projectFolder, RushConstants.packageDepsFilename);
       if (fsx.existsSync(currentDepsPath)) {
-        lastPackageDeps = JSON.parse(fsx.readFileSync(currentDepsPath, 'utf8')) as IPackageDeps;
+        lastPackageDeps = JSON.parse(fsx.readFileSync(currentDepsPath, 'utf8')) as IPackageDependencies;
```

Comment:
>JSON.parse [](start = 26, length = 10)

Use JsonFile.readJsonFile(), which uses JJU to give professional quality error messages.